vassal-app/src/main/java/VASSAL/counters/Hideable.java

```diff
@@ -210,14 +211,41 @@ public void draw(Graphics g, int x, int y, Component obs, double zoom) {
         g2d.fill(t.createTransformedShape(piece.getShape()));
       }
 
-      final Composite oldComposite = g2d.getComposite();
-      g2d.setComposite(
-        AlphaComposite.getInstance(AlphaComposite.SRC_OVER, transparency)
-      );
-      piece.draw(g, x, y, obs, zoom);
-      g2d.setComposite(oldComposite);
+      // 1) Determine piece bounds at current zoom
+      final Rectangle bounds = piece.getShape().getBounds();
+      final int w = (int) Math.ceil(bounds.width * zoom);
+      final int h = (int) Math.ceil(bounds.height * zoom);
+
+      // If there's nothing visible at this zoom, skip
+      if (w <= 0 || h <= 0) {
+        return;
+      }
+
+      // 2) Create offscreen image and draw piece fully opaque
+      final BufferedImage offscreen = new BufferedImage(w, h, BufferedImage.TYPE_INT_ARGB);
+      final Graphics2D offG = offscreen.createGraphics();
+      try {
+        offG.scale(zoom, zoom);
+        offG.translate(-bounds.x, -bounds.y);
+
+        piece.draw(offG, 0, 0, obs, 1.0); // Fully opaque
+      }
+      finally {
+        offG.dispose();
+      }
```

Comment:
This is going to cause a `BufferedImage` to be created for each invisible-to-others `Hideable` on each piece on each repaint. I am concerned about the performance impact of this.

Reply:
The `BufferedImage` you're creating here is also not guaranteed to be the preferred image type for the graphics device it will be painted to.

Reply:
You can use `VASSAL.tools.image.ImageUtils.createCompatibleTranslucentImage()` to make a `BufferedImage` with an alpha channel of the right image type for the frame buffer. To get the image into the image cache so that it's not recreated on each draw might require using `GamePieceOp`.

But I'm not convinced this is the right approach to begin with.

Reply:
Would it make more sense to keep the BufferedImage around ("cache it") after it gets created the first time, so that it isn't getting re-generated every frame? So the first time this draw path comes up it makes the BufferedImage and then after that it just keeps it around?